### PR TITLE
Adds the repack job for repackaging k8s releases

### PR DESCRIPTION
**Pull request by lazypower**

### CODE REVIEW

File: kubernetes-upstream-repack/config.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        <artifactDaysToKeep>-1</artifactDaysToKeep>
+        <artifactNumToKeep>-1</artifactNumToKeep>
+      </strategy>
+    </jenkins.model.BuildDiscarderProperty>
+    <hudson.model.ParametersDefinitionProperty>
+      <parameterDefinitions>
+        <hudson.model.StringParameterDefinition>
+          <name>KUBERNETES_RESOURCE</name>
+          <description>Kubernetes Release tarball http/https url</description>
+          <defaultValue>https://github.com/kubernetes/kubernetes/releases/download/v1.4.4/kubernetes.tar.gz</defaultValue>
```
Comment (mbruzek) [inline kubernetes-upstream-repack/config.xml]: Could we build the URL with the $VERSION variable so it appears like the other builders?

File: kubernetes-upstream-repack/config.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+if [ -f kubernetes.tar.gz ]; then
+  rm -rf kubernetes.tar.gz
+fi
+
+wget $KUBERNETES_RESOURCE -o kubernetes.tar.gz
+
+ls -al
+
+./package_kubernetes.sh $PWD/kubernetes.tar.gz.1</command>
```
Comment (mbruzek) [inline kubernetes-upstream-repack/config.xml]: Is this file name "kuberentes.tar.gz.1" correct here or is this a duplicate artifact of wget?

File: kubernetes-upstream-repack/config.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        </hudson.model.StringParameterDefinition>
+      </parameterDefinitions>
+    </hudson.model.ParametersDefinitionProperty>
+  </properties>
+  <scm class="hudson.plugins.git.GitSCM" plugin="git@3.0.0">
+    <configVersion>2</configVersion>
+    <userRemoteConfigs>
+      <hudson.plugins.git.UserRemoteConfig>
+        <url>git@github.com:chuckbutler/container-utils.git</url>
+        <credentialsId>jenkins_github_credential</credentialsId>
```
Comment (mbruzek) [inline kubernetes-upstream-repack/config.xml]: We can make the repo public if that is the only reason for your credentials here.

File: kubernetes-upstream-repack/config.xml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>chmod +x package_kubernetes.sh
+
+if [ -f kubernetes.tar.gz ]; then
+  rm -rf kubernetes.tar.gz
+fi
+
+wget $KUBERNETES_RESOURCE -o kubernetes.tar.gz
```
Comment (mbruzek) [inline kubernetes-upstream-repack/config.xml]: We should build the URL here from the $VERSION variable if possible.

**Review by mbruzek — CHANGES_REQUESTED** [review]
I would like to see some changes in this request. Let me know if you disagree with my suggestions.

File: kubernetes-upstream-repack/config.xml (hunk 2 above)
Comment (lazypower) [reply kubernetes-upstream-repack/config.xml]: no, its kubernetes.tar.gz.1, and it baffled me... i think whats happening is wget is following the redirect and its getting both the html document and the package deliverable.

File: kubernetes-upstream-repack/config.xml (hunk 3 above)
Comment (lazypower) [reply kubernetes-upstream-repack/config.xml]: thats fine, but it needs to happen before i remove credentials from the job. chicken/egg.

File: kubernetes-upstream-repack/config.xml (hunk 1 above)
Comment (lazypower) [reply kubernetes-upstream-repack/config.xml]: certainly

**Comment by mbruzek** [comment]
+1 This looks more like the others.

**Pull request merged**